module/reports/helper.js

```diff
@@ -0,0 +1,557 @@
+
+            } catch (error) {
+                return reject(error);
+            }
+        })
+    }
+
+    /**
+    * Detail view Report.
+    * @method
+    * @name detailView 
+    * @param {String} userId - Logged in user id.
+    * @param {String} userToken - Logged in user keyclock token.
+    * @param {String} reportType - report type monthly or quterly.
+    * @param {String} entityId - mapped entity id.
+    * @param {String} programId - program id
+    * @param {Boolean} getPdf - to get pdf 
+    * @returns {Object} - response consist of chart report data
+   */
+
+    static detailView(userId, userToken, reportType, entityId = "", programId, getPdf) {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                let query = {
+                    isDeleted: { $ne: true }
+                };
+
+                if (entityId) {
+                    query["entityInformation._id"] = ObjectId(entityId);
+                } else {
+                    query["userId"] = userId
+                }
+
+                let chartObject = [];
+                let dateRange = await _getDateRangeofReport(reportType);
+                let endOf = dateRange.endOf;
+                let startFrom = dateRange.startFrom;
+
+                query['$or'] = [
+                    { "syncedAt": { $gte: new Date(startFrom), $lte: new Date(endOf) } },
+                    { "tasks": { $elemMatch: { syncedAt: { $gte: new Date(startFrom), $lte: new Date(endOf) } } } },
+                ]
+
+                if (programId) {
+                    query['programInformation._id'] = ObjectId(programId);
+                }
+
+                const projectDetails = await userProjectsHelper.projectDocument(
+                    query,
+                    ["title",
+                        "taskReport",
+                        "status",
+                        "tasks.status",
+                        "tasks.startDate",
+                        "tasks.endDate",
+                        "tasks.name",
+                        "categories",
+                        "startDate",
+                        "endDate"],
+                    []
+                );
+
+                if (!projectDetails.length > 0) {
+
+                    return resolve({
+                        message: CONSTANTS.apiResponses.REPORTS_DATA_NOT_FOUND,
+                        result: []
+                    })
+                }
+
+                if (getPdf == true) {
+
+                    let types = await this.types();
+                    let returnTypeInfo = types.result.filter(type => {
+                        if (type.value == reportType) {
+                            return type.label;
+                        }
+                    });
+
+                    let projectData = [];
+                    await Promise.all(projectDetails.map(project => {
+                        let projectInfo = {
```

Comment:
@rakeshSgr - Why go for the map as it is the mongo document is in the required format only